app/api/consent_type_endpoint.py

```diff
@@ -0,0 +1,59 @@
+from fastapi import APIRouter, Depends, HTTPException
+from app.api.auth.authentication import get_current_user
+from app.config import server
+from pydantic import BaseModel, validator
+from uuid import UUID
+from tracardi.service.storage.driver import storage
+from elasticsearch import ElasticsearchException
+
+
+class ConsentForm(BaseModel):
+    id: UUID
+    name: str
+    description: str
+    revokable: bool
+    default_value: str
+
+    @validator("default_value")
+    def default_value_validator(cls, v):
+        if v not in ("grant", "deny"):
+            raise ValueError("'default_value' must be either 'grant' or 'deny'")
+        return v
+
+
+router = APIRouter(
+    dependencies=[Depends(get_current_user)]
+)
+
+
+@router.post("/consent", tags=["consent"], include_in_schema=server.expose_gui_api, response_model=dict)
+async def add_consent_type(data: ConsentForm):
```

Comment:
Do not require id from user. generate it before saving. remove id form ConsentForm (ConsentTypePayload) and make new object ConsentType and save this object. Payload is something different then ConsentType. Making 2 objects makes it possible to change ConsentType without changing API.